teachertool/src/components/CatalogModal.tsx

```diff
@@ -0,0 +1,71 @@
+/// <reference path="../../../built/pxtblocks.d.ts"/>
+
+import { useContext, useState } from "react";
+import { AppStateContext } from "../state/appStateContext";
+import { Checkbox } from "react-common/components/controls/Checkbox";
+import { Modal } from "react-common/components/controls/Modal";
+import { hideCatalogModal } from "../transforms/hideCatalogModal";
+import { addCriteriaToRubric } from "../transforms/addCriteriaToRubric";
+
+interface IProps {}
+
+const CatalogModal: React.FC<IProps> = ({}) => {
+    const { state: teacherTool, dispatch } = useContext(AppStateContext);
+    const [ checkedCriteriaIds, setCheckedCriteria ] = useState<string[]>([]);
+
+    function handleCheckboxChange(criteria: pxt.blocks.CatalogCriteria, newValue: boolean) {
```

Comment:
Something I was thinking about.. What if we added a `selected` field to the Catalog Criteria Schema? Since we parse the catalog from docs into state, we can manage whether or not something is checked from the catalog directly. The look-up for an object like that would be faster, right? And then when the checkbox is changed, we can just toggle that field. The done and close modal would have to be modified, though. Also, I'm not a huge fan of the implications that this has on state.. it feels kinda icky that a modal would have so much power over the state of the catalog, but I also feel like this handler should be simpler than having to do a look-up or filter.

Reply:
We want the user to be able to have multiple instances of the same catalog criteria, so just having `selected` on catalog criteria doesn't quite work (if it did, there'd be no real need for criteria instances at all).

I'm not sure I quite understand the concern around the handler. Could you elaborate on that a bit?

Reply:
Yeah, my idea would be that the `selected` state on the catalog would be reset every time the modal is closed so when the user re-opens the modal, they have a fresh state. This might be more work than it's worth. 

In regards to the handler, I'm a bit worried about a list operation every time the box is checked or unchecked. To be fair, they shouldn't be very computationally taxing, but the list operations plus conditions when we want to toggle a condition for something seems like a lot of logic.

Reply:
Per our discussion in person, I think something like that might work (we'd need a new component like a CriteriaCheckbox to store `selected`, since I don't think we want UI-related fields to leak out into non-UI-specific classes), but it would probably just move the complexity elsewhere rather than actually simplifying things, so I'm inclined to leave this as-is at the moment.